drools-ansible-rulebook-integration-api/src/test/java/org/drools/ansible/rulebook/integration/api/MultipleRuleMatchTest.java

```diff
@@ -248,4 +248,399 @@ private static void checkPartialMatchesWithMatchMultipleRules(boolean matchMulti
+                        "                                        }\n" +
+                        "                                    }\n" +
+                        "                                }\n" +
+                        "                            ]\n" +
+                        "                        },\n" +
+                        "                        \"actions\": [\n" +
+                        "                            {\n" +
+                        "                                \"Action\": {\n" +
+                        "                                    \"action\": \"debug\",\n" +
+                        "                                    \"action_args\": {\n" +
+                        "                                        \"msg\": \"Fourth one matches\"\n" +
+                        "                                    }\n" +
+                        "                                }\n" +
+                        "                            }\n" +
+                        "                        ],\n" +
+                        "                        \"enabled\": true\n" +
+                        "                    }\n" +
+                        "                }\n" +
+                        "            ]\n" +
+                        "        }";
+
+        RulesExecutor rulesExecutor = RulesExecutorFactory.createFromJson(rules);
+
+        List<Match> matchedRules = rulesExecutor.processEvents("{ \"i\" : 0 }").join();
+        assertThat(matchedRules).hasSize(1);
+        assertThat(matchedRules.stream().map(m -> m.getRule().getName())).contains("R1");
+
+        matchedRules = rulesExecutor.processEvents("{ \"j\" : 0 }").join();
+        assertThat(matchedRules).hasSize(1);
+
+        // when multiple match is allowed i=0 should be retained and now used to also fire R2
+        assertThat(matchedRules.stream().map(m -> m.getRule().getName())).contains("R2");
+        // i=0 and j=0 have a partial match for R3. Retained
+        assertThat(rulesExecutor.getAllFacts()).hasSize(2);
+
+        matchedRules = rulesExecutor.processEvents("{ \"k\" : 0 }").join();
+        assertThat(matchedRules).hasSize(1);
+
+        // when multiple match is allowed i=0 and j=0 should be retained and now used to also fire R3
+        assertThat(matchedRules.stream().map(m -> m.getRule().getName())).contains("R3");
+        // i=0 and j=0 have a partial match for R3. Retained. But k=0 is discarded
+        assertThat(rulesExecutor.getAllFacts()).hasSize(2);
+
+        matchedRules = rulesExecutor.processEvents("{ \"l\" : 0 }").join();
+        assertThat(matchedRules).hasSize(1);
+
+        // when multiple match is allowed i=0 and j=0 should be retained and now used to also fire R4
+        assertThat(matchedRules.stream().map(m -> m.getRule().getName())).contains("R4");
+        // if all rules fired now the working memory should be empty
+        assertThat(rulesExecutor.getAllFacts()).isEmpty();
```

Comment:
Fails here, because `i=0` and `j=0` are still in working memory.

With a Debugger, I see that the first peer (= JoinNodeLT[12]) didn't have a child (It should be RTNLT of R3)  when checking the partial match for R4 firing.